ba.bitcamp.guihomework/src/task4/JustWindow.java

```diff
@@ -0,0 +1,64 @@
+package task4;
+
+import javax.swing.JOptionPane;
+
+/**
+ * This program calculate the sum of all numbers user has input.
+ */
+public class JustWindow {
+
+	public static void main(String[] args) {
+
+		int sum = 0;
+
+		int choice;
+/**
+ * We use this loop to see are they equals and to catch all possible exceptions.
+ */
+		do {
+
+			String b = JOptionPane.showInputDialog("Enter a number");
+
+			try {
+
+				if (!(b.equals(null))) {
+
+					int a = Integer.parseInt(b);
+
+					sum += a;
+
+				}
+
+			} catch (NullPointerException ex) {
```

Comment:
NPE se nikad u principu ne hvata, vec se treba razmisljati o tome da se on nikad ne desi, a ako se desi, to je runtime exception koji ce prekinuti izvrsavanje programa. Dakle, ispravan pristup je provjeriti da li je nesto null i u skladu sa tim izvrsiti odredjenu akciju.